ui/src/components/admin/fundraise/AdminFundraiseContainer.js

```diff
@@ -0,0 +1,326 @@
+    }
+  };
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    if (!campaignName || campaignName.length === 0 || campaignNameCheck())
+      return;
+
+    setIsLoading(true);
+    let success = false;
+    if (id && id.length > 0) {
+      success = await updateHandler();
+    } else {
+      success = await createHandler();
+    }
+
+    if (success) {
+      toast({
+        description: 'Saved',
+        status: 'success',
+        duration: 5000,
+      });
+      await getData();
+    }
+    setIsLoading(false);
+  };
+  const resetHandler = () => {
+    setId('');
+    setCampaignName('');
+    setCategoryName('');
+    setCategoryKey('');
+    setAmount('');
+    setGivers('');
+    setMilestones('');
+    setSelected();
+  };
+  const campaignNameCheck = () => {
+    if (
+      fundraises &&
+      fundraises.some((i) => i.id !== id && i.campaignName === campaignName)
+    )
+      return true;
+
+    return false;
+  };
+  const handleAddMilestones = () => {
+    if (milestoneName && milestoneAmount && milestoneDeadline) {
+      const newMilestone = {
+        milestoneName,
+        milestoneAmount,
+        milestoneDeadline,
+      };
+      setMilestones([...milestones, newMilestone]);
+      setMilestoneName('');
+      setMilestoneAmount('');
+      setMilestoneDeadline('');
+    }
+  };
+
+  const handleRemoveMilestones = (index) => {
+    const updatedMilestones = [...milestones];
+    updatedMilestones.splice(index, 1);
+    setMilestones(updatedMilestones);
+  };
+
+  return (
+    <Container w="100%" maxW="100%">
+      <Heading as="h5" mb={5}>
+        Fund Raising Manager
+      </Heading>
+      <Stack direction={['column', 'row']} w="100%">
+        <Box w={['100%', '50%']}>
+          <form onSubmit={onSubmit}>
+            <FormControl isRequired isInvalid={campaignNameCheck()}>
+              <FormLabel>Campaign Name</FormLabel>
+              <Input
+                type="text"
+                value={campaignName}
+                onChange={(e) => setCampaignName(e.target.value)}
+              />
+              <FormErrorMessage>
+                Another fundraising campaign with the same name already exists
+              </FormErrorMessage>
+            </FormControl>
+            <FormControl isRequired>
+              <FormLabel>Category Name</FormLabel>
+              <Input
+                type="text"
+                value={categoryName}
+                onChange={(e) => setCategoryName(e.target.value)}
+              />
+            </FormControl>
+            <FormControl isRequired>
+              <FormLabel>Category Key</FormLabel>
+              <Input
+                type="text"
+                value={categoryKey}
+                onChange={(e) => setCategoryKey(e.target.value)}
+              />
+            </FormControl>
+            <FormControl isRequired>
+              <FormLabel>Amount</FormLabel>
+              <NumberInput>
+                <NumberInputField
```

Comment:
<img width="945" alt="image" src="https://github.com/hmcc-global/hmcchk-web/assets/56828805/8d6fdbbc-4ee7-466e-9add-aff3522c55f4">
I can't get the numbers to show here. But the function updates the numbers properly in ag-grid. UI issue?

Reply:
was ui issue due to <NumberInput>, turns out it cant get values the same way as normal <Input>.
switched to <Input type = 'number'> works now